### ISSUE TITLE
poly function giving an ArgumentError: range step cannot be zero

### ISSUE THREAD

**Issue by mmkim1210** [MakieOrg/Makie.jl#1579]
Hello, I would like to plot a straightforward polygon as below, but I keep getting an error, `ERROR: ArgumentError: range step cannot be zero`. The entire stacktrace is pretty long, so I will not attach it here, but I am running the latest version of Julia and Makie. 
```julia
poly(Point2f[(2.32756952e8, 0.9), (2.32756952e8, 1), (2.32757008e8, 1), (2.32757008e8, 0.9)])
```

**Comment by ffreyer** [comment]
This is another version of #931. You might be able to fix it by explicitly setting x limits before plotting, though the poly might have issues rendering as well. The other option is to shift this close to 0 and adjust the axis labels.

**Comment by asinghvi17** [comment]
This should be fixed now with v0.21.
![iTerm2 tKyaAC](https://github.com/MakieOrg/Makie.jl/assets/32143268/b3023d48-796e-4920-950b-58d16478dc61)
